store/view/articles.js

```diff
@@ -0,0 +1,168 @@
+
+    // Inject articles and return their codes
+    return dispatch('resources/injectResource', {
+      path: 'articles',
+      data: await dispatch('prepareArticles', data)
+    }, { root: true })
+  }),
+  getMainlist: composer.compose({
+    before: [
+      compounds.doSth(({ ctx }) => ctx.commit(changeCurrCategory('wszystkie'))),
+      compounds.callActionWhen(({ ctx }) => _.isEmpty(ctx.getters.mainlist)),
+      compounds.doSth(({ ctx }) => ctx.commit(loading(true)))
+    ],
+    caller: callers.squashErrorsCall('#getMainlist'),
+    success: [
+      opuses.doSth(({ ctx, result }) => ctx.dispatch('addCodes', { path: 'mainlist', codes: result })),
+      opuses.doSth(({ ctx, result }) => ctx.dispatch('addCodes', { path: 'all', codes: result })),
+      opuses.doSth(({ ctx, result }) => ctx.commit(loading(false)))
+    ]
+  })(async function getMainlistAction ({ state, commit, dispatch }, params) {
+    const data = await knitService.getCollection(this.$axios, 'articles', settings.defaultQsObject)
+
+    // Inject articles and return their codes
+    return dispatch('resources/injectResource', {
+      path: 'articles',
+      data: await dispatch('prepareArticles', data)
+    }, { root: true })
+  }),
+  getArticle: composer.compose({
+    before: [
+      compounds.callActionWhen(({ ctx, params }) => !_.includes(ctx.getters['all'], params.code)),
+      compounds.doSth(({ ctx }) => ctx.commit(loading(true)))
+    ],
+    caller: callers.squashErrorsCall('#getArticle'),
+    success: [
+      opuses.doSth(({ ctx, result }) => ctx.dispatch('addCodes', { path: 'all', codes: result })),
+      opuses.doSth(({ ctx, result }) => ctx.commit(loading(false)))
+    ],
+    always: [
+      opuses.doSth(({ ctx, params }) => params.code !== ctx.getters['currArticleCode'] && ctx.commit(changeCurrArticle(params.code)))
+    ]
+  })(async function getArticleAction ({ commit, dispatch }, params) {
+    const data = await knitService.getCollection(this.$axios, 'articles', {
+      ...settings.defaultQsObject,
+      code: params.code,
+      limit: 1
+    })
+
+    // Inject articles and return their codes
+    return dispatch('resources/injectResource', {
+      path: 'articles',
+      data: await dispatch('prepareArticles', data)
+    }, { root: true })
+  }),
+  prepareArticles (ctx, data) {
+    const articlesArr = commonHelper.pickItemsProps(data['hydra:member'], settings.props, true, settings.datePicker)
+    return _.keyBy(articlesArr, article => article.code)
+  },
+  addCodes ({ commit }, params) {
+    knitLogger.debug(() => `Adding codes to 'all' and '${params.path}': ${JSON.stringify(params.codes)}`)
+    commit(codes(params))
+  }
+}
+
+export const mutations = {
+  [ types.LOAD_ARTICLES ] (state, payload) {
+    state.loading = payload.loading
+  },
+  [ types.ADD_CODES ] (state, payload) {
+    if (payload.path) {
+      state[payload.path] = _.union(state[payload.path], payload.codes)
+    } else {
```

Comment:
I don't really like this one (having two different behaviours), is there something we could do @FieryCod ? Like split to TWO mutations? or it would be to much impractical?

Reply:
Well, I do not like it either. But I do not know whether having two mutations like
`ADD_CODES_CATEGORY` or `ADD_CODES_TO_STH` makes sense here. I don't know. @zrebcu411 could you propose some solution maybe?

Reply:
I think it's ok :)